Julian leung/custom reporting documentation (#7116)
DataDog/documentation: content/en/account_management/billing/custom_usage_reporting.md

---
title: Custom usage reporting
kind: documentation
---

## Overview

> Administrators can access the Custom Reporting tab from the Plan & Usage section in Datadog, then navigate to: 

Reviewer: ```suggestion
Administrators can access the Custom Reporting tab from the Plan & Usage section in Datadog. Navigate to: 
```

> `Plan & Usage`--> `Custom Reporting`.

Reviewer: ```suggestion
`Plan & Usage` > `Custom Reporting`.
```

> This tool enables organizations to split their Datadog usage by up to 3 chosen tag keys and generates daily and monthly usage reports.

Reviewer: ```suggestion
This tool enables organizations to split their Datadog usage by up to three chosen tag keys. This tool also generates daily and monthly usage reports.
```

The Custom Reporting page provides the following information and functionality:

* It lists the existing tag keys that usage is being broken down by and provides the ability to change and add new ones (up to 3 tag keys).
* It generates daily .tsv (tab separated values) files for most usage types.
* It summarizes usage at the end of each month and surfaces both in the UI and as a .tsv download.

_Note_: The following usage types are not supported in this tool yet. They will be added subsequently:

* Indexed Log Events

> * Ingested Log Events

Reviewer 2: Let's make it `Ingested Logs`  instead of `Ingested Log Events`

* Analyzed Spans

### Getting Started

In order to start receiving daily data, an administrator will need to create a new report via the user interface. 

{{< img src="account_management/billing/advanced-usage-reporting-01.png" alt="Getting Started" >}}

The 'Applied Tags' section enables the following:

> * Entering up to three tag keys from a dropdown. The dropdown is pre-populated with existing tags on the account.

Reviewer 2: `Entering up to three tag keys from a dropdown. The dropdown is pre-populated with existing tags on both the root account and any child organizations under the account.`

Add that last sentence.

### Interpreting the Data

The table below shows a sample daily report for Custom Metrics usage two tags: `team` and `service`.

| public_id   | hour                  | team      | service               | num_custom_timeseries           |
|-------------|-----------------------|-----------|-----------------------|---------------------------------|
| publicid1   | 2020-02-01 00:00:00   | &lt;empty&gt; | service1 &#124; service2 | 50                              |
| publicid1   | 2020-02-01 09:00:00   | team1     |                       | 28                              |
| publicid1   | 2020-02-01 18:00:00   | team2     | service3              | 1023                            |

* An `<empty>` value means the resource was tagged with the respective tag but did not have a value.
* No value means the resource was not tagged with that particular tag (e.i. N/A).
* | (pipe) separated values (e.i. `service1 | service2`) mean that a particular tag was applied multiple times on the resource.
* A valid tag value (as described [here][1]) refers to the actual value of the respective tag.

[1]: https://docs.datadoghq.com/tagging/#defining-tags

> [2]: https://docs.datadoghq.com/api/#get-daily-custom-reporting-available-files

Reviewer 3: Just a proposed addition (in need of some rewording):

```
## Further / advanced data analysis

When using multiple tags both the Daily or Monthly Custom reports will contain data for all possible combinations of those tags. Which makes them good candidates for use as base datasets to further data analysis tasks. For instance, by grouping or pivoting, to produce views focused on one or some of the tags set, or to do aggregations across custom date ranges.```

Reviewer 3: cc @wissallahjouji ^^

Reviewer: reworded:
> ## Further data analysis
> When using multiple tags, both the Daily and Monthly Custom reports contain data for all possible combinations of those tags, and are suitable to use as base datasets for further data analysis tasks. For instance, you can use grouping or pivoting to produce views focused on a subset of the tags, or to perform aggregations across custom date ranges.
